autorag/data/corpus/llama_index.py

```diff
@@ -60,10 +66,22 @@ def llama_text_node_to_parquet(text_nodes: List[TextNode],
     corpus_df = pd.DataFrame(list(map(lambda node: {
         'doc_id': node.node_id,
         'contents': node.text,
-        'metadata': add_essential_metadata(node.metadata)
+        'metadata': add_essential_metadata_llama_text_node(node.metadata, node.relationships)
     }, text_nodes)))
 
     if output_filepath is not None:
         save_parquet_safe(corpus_df, output_filepath, upsert=upsert)
 
     return corpus_df
+
+
+def add_essential_metadata_llama_text_node(metadata: Dict, relationships: Dict) -> Dict:
+    if 'last_modified_datetime' not in metadata:
+        metadata['last_modified_datetime'] = datetime.now()
+    prev_node = relationships.get(NodeRelationship.PREVIOUS, None)
+    if prev_node:
+        metadata['prev_id'] = prev_node.node_id
+    next_node = relationships.get(NodeRelationship.NEXT, None)
+    if next_node:
+        metadata['next_id'] = next_node.node_id
+    return metadata
```

Comment:
I think adding `prev_id` and `next_id` to the `validate_corpus_data` function just for the `prev_next_augmenter` is overkill.

So I created a `make_prev_next_id` function in `cast_corpus_dataset` to force the `prev_id` and `next_id` to be added (like 'last_modified_datatime').


In `llama_index_text_node_to_parquet`, if there is no `relations`, skip it, and in `cast_corpus_dataset`, add `prev_id` and `next_id`. 
(Since we're going through `cast_corpus_dataset` before using `prev_next_augmenter` anyway + decided it was unnecessarily long code. 
Instead, I added logic to not throw an error if relationships is None)